mozdownload/scraper.py

```diff
@@ -332,10 +335,15 @@ def show_matching_builds(self, builds):
 class DailyScraper(Scraper):
     """Class to download a daily build from the Mozilla server."""
 
-    def __init__(self, branch='mozilla-central', build_id=None, date=None,
+    def __init__(self, branch=None, build_id=None, date=None,
                  build_number=None, revision=None, *args, **kwargs):
         """Create an instance of the daily scraper."""
-        self.branch = branch
+        if not branch:
+            application = kwargs.get('application', None)
```

Comment:
Sorry for the late response. I tried what you've said, it's again resulting in some other error.
```
self = <test_daily_indices.TestDailyScraperIndices testMethod=test_build_indices>

    def test_build_indices(self):
        """Testing for correct build_index in DailyScraper"""
    
        for entry in test_params:
            scraper = DailyScraper(destination=self.temp_dir,
                                   base_url=self.wdir,
                                   logger=self.logger,
                                   **entry['args'])
>           self.assertEqual(scraper.build_index, entry['build_index'])
E           AttributeError: 'DailyScraper' object has no attribute 'build_index'

tests/daily_scraper/test_daily_indices.py:54: AttributeError
```